src/components/Faucet.tsx

```diff
@@ -0,0 +1,178 @@
+"use client";
+
+import React, { useState } from 'react';
+import { Card, CardContent } from "@/components/ui/card";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { BigNumber, ethers } from 'ethers';
+
+const Faucet = () => {
+    const [verification, setVerification] = useState(false);
+    const [userAddress, setUserAddress] = useState('');
+    const [isValidAddress, setIsValidAddress] = useState(true);
+    const [txHash, setTxHash] = useState<string | null>(null);
+    const [lastWithdrawalTime, setLastWithdrawalTime] = useState<BigNumber | null>(null);
+    const [withdrawalDelay, setWithdrawalDelay] = useState<BigNumber | null>(null);
+
+    const toastId = React.useRef<string | number | null>(null);
+
+    const checkEligibility = async () => {
+        toastId.current = toast.loading("Verifying Address...");
+        const response = await fetch('/api/faucet/verify', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({ address: userAddress })
+        });
+        const result = await response.json();
+
+        if (response.ok && result.message === 'Address is allowed to withdraw') {
```

Comment:
instead of using the message to check, use response codes

Send 200 status code from the backend
and send the message. The message can be used directly in the toast

Similarly, if the error code is 400 from the backend, send the message "you are not whitelisted" and use that in the toast directly